test/api/user/post.users.spec.ts

```diff
@@ -0,0 +1,119 @@
+            description: 'A versatile back-end node.js developer',
+        })
+
+        expect(res.status).to.be.equal(400)
+    })
+
+    it('should fail to create a new user with missing dob', async () => {
+        const res = await request(testServer).post(endpoint.route).send({
+            name: 'Emmanuel',
+            email: 'emmanuel@test.com',
+            password: 'AAaa@@88$$99',
+            address: '44-65 Laparella Cinco, Donella, Mexico City, Mexico',
+            description: 'A versatile back-end node.js developer',
+        })
+
+        expect(res.status).to.be.equal(400)
+    })
+
+    it('should fail to create a new user with missing password', async () => {
+        const res = await request(testServer).post(endpoint.route).send({
+            name: 'Emmanuel',
+            email: 'emmanuel@test.com',
+            dob: '1996-05-29',
+            address: '44-65 Laparella Cinco, Donella, Mexico City, Mexico',
+            description: 'A versatile back-end node.js developer',
+        })
+
+        expect(res.status).to.be.equal(400)
+    })
+
+    it('should fail to create a new user with password that does not meet standards', async () => {
+        const res = await request(testServer).post(endpoint.route).send({
+            name: 'Emmanuel',
+            email: 'emmanuel@test.com',
+            dob: '1996-05-29',
+            password: 'AAaaretrtret',
+            address: '44-65 Laparella Cinco, Donella, Mexico City, Mexico',
+            description: 'A versatile back-end node.js developer',
+        })
+
+        expect(res.status).to.be.equal(400)
+    })
+
+    it('should create a new user', async () => {
+        const res = await request(testServer).post(endpoint.route).send({
+            name: 'Emmanuel',
+            email: 'emmanuel@test.com',
+            dob: '1996-05-29',
+            password: 'AAaa@@88$$99',
+            address: '44-65 Laparella Cinco, Donella, Mexico City, Mexico',
+            description: 'A versatile back-end node.js developer',
+        })
+
+        expect(res.status).to.be.equal(201)
+        expect(res.body).to.be.jsonSchema(userSchema)
+    })
+
+    it('should fail to create a user who already exists', async () => {
+        const users = await generateDatabaseUsers(database, 1)
+        const res = await request(testServer).post(endpoint.route).send({
+            name: users[0].name,
+            email: users[0].email,
+            dob: '1996-05-29',
+            password: 'AAaa@@88$$99',
+        })
+
+        expect(res.status).to.be.equal(409)
+    })
+
+    it('should create a new user without address or description', async () => {
+        const res = await request(testServer).post(endpoint.route).send({
+            name: 'Mina',
+            email: 'mina@test.com',
+            dob: '1996-05-29',
+            password: 'AAaa@@88$$99',
+        })
+
+        expect(res.status).to.be.equal(201)
+        expect(res.body).to.be.jsonSchema(userSchema)
```

Comment:
can we assert the result from a more business oriented perspective, not only verify the syntax of the response, but also verify that the user is saved in the system that can be retrieved later?

Reply:
Yes, this is good practice. Asserting the syntax only is no assurance that the document is saved in the database. I missed implementing this assertion here. 
I actually do this for the `delete /user/:id` endpoint. I assert that that the document does not exist in the database. However, it should also be done for endpoints that create and update resources such as this to ensure the resource was actually saved/updated in the database I can update the test code.